dali/pipeline/data/sample_view.h

```diff
@@ -182,14 +185,12 @@ class ConstSampleView : public SampleViewBase<Backend, const void *> {
   }
 
   ConstSampleView &operator=(SampleView<Backend> &&other) {
-    if (this != &other) {
-      data_ = other.data_;
-      other.data_ = nullptr;
-      shape_ = std::move(other.shape_);
-      other.shape_ = {0};
-      type_id_ = other.type_id_;
-      other.type_id_ = DALI_NO_TYPE;
-    }
+    data_ = other.data_;
+    other.data_ = nullptr;
+    shape_ = std::move(other.shape_);
+    other.shape_ = {0};
+    type_id_ = other.type_id_;
+    other.type_id_ = DALI_NO_TYPE;
```

Comment:
You might consider using `std::exchange` for all of those

Reply:
I will look into a separate PR with that